tests/integration/pipeline/test_bootstrap_command.py

```diff
@@ -0,0 +1,229 @@
+from unittest import skipIf
+from unittest.mock import Mock
+
+from parameterized import parameterized
+
+from samcli.lib.pipeline.bootstrap.stage import Stage
+from tests.integration.pipeline.base import BootstrapIntegBase
+from tests.testing_utils import (
+    run_command_with_input,
+    RUNNING_ON_CI,
+    RUNNING_TEST_FOR_MASTER_ON_CI,
+    RUN_BY_CANARY,
+    run_command,
+)
+
+# bootstrap tests require credentials and CI/CD will only add credentials to the env if the PR is from the same repo.
+# This is to restrict tests to run outside of CI/CD, when the branch is not master or tests are not run by Canary
+SKIP_DEPLOY_TESTS = RUNNING_ON_CI and RUNNING_TEST_FOR_MASTER_ON_CI and not RUN_BY_CANARY
+
+
+@skipIf(SKIP_DEPLOY_TESTS, "Skip bootstrap tests in CI/CD only")
+class TestBootstrap(BootstrapIntegBase):
+    @parameterized.expand([("create_ecr_repo",), (False,)])
+    def test_interactive_with_no_resources_provided(self, create_ecr_repo: bool):
+        stage_name = self._method_to_stage_name(self.id())
+        mock_stage = Mock()
+        mock_stage.name = stage_name
+        self.stack_names = [Stage._get_stack_name(mock_stage)]
```

Comment:
this logic is repeated in many place. Can we have something like
`stage_name, stack_name = self._method_to_stage_and_stack_names()` ? 
Also, we can create real Stage instead of mock? the only required attribute is stage_name

Reply:
yeah good idea, updated